src/lesson5/javaBeginnerChapter2/Task21.java

```diff
@@ -0,0 +1,59 @@
+package lesson5.javaBeginnerChapter2;
+
+import java.util.Scanner;
+
+//Напишите три цикла выполняющих многократное сложение строк, один с
+//помощью оператора сложения и String, другой с помощью StringBuilder и метода
+//append, а также аналогино для StringBuffer. Сравните скорость их выполнения.
+public class Task21 {
+    public static void main(String[] args) {
+        Scanner sc = new Scanner(System.in);
+        System.out.print("Enter string(class String): ");
+        String str = sc.nextLine();
+        String newText = "";
+        str = str.trim().replaceAll(" +", " ");
+        if (str.length() != 0) {
+            long start = System.nanoTime();
+            for (int i = 0; i < str.length(); i++) {
+                if (str.charAt(i) != ' ') {
+                    newText += str.charAt(i);
+                }
+            }
+            long end = System.nanoTime();
+            System.out.println("String: " + newText);
+            System.out.println("Total execution time: " + (end - start));
+        }
+
+        System.out.print("Enter string(class StringBuilder): ");
+        StringBuilder str2 = new StringBuilder(sc.nextLine());
+        StringBuilder newText2 = new StringBuilder();
+        if (str2.length() != 0) {
+            long start = System.nanoTime();
+            for (int i = 0; i < str2.length(); i++) {
+                if (str2.charAt(i) != ' ') {
+                    newText2.append(str2.charAt(i));
+                }
+            }
+            long end = System.nanoTime();
+            System.out.println("String: " + newText2);
+            System.out.println("Total execution time: " + (end - start));
+        }
+        System.out.println(newText2);
+
+        System.out.print("Enter string(class StringBuffer): ");
+        StringBuffer str3 = new StringBuffer(sc.nextLine());
```

Comment:
What is difference between string builder and string buffer? Why does IDEA suggest in this place to use string builder?

Reply:
StringBuffer is synchronized i.e. thread safe. StringBuilder is non-synchronized i.e. not thread safe.

Reply:
It's just words from the book) Our current logic doesn't contain multithreading, so no sense to use StringBuffer (IDEA shows it in the hint). All synchronized methods and collections will work same or longer than their non-synchronized analogs. So it we compare StringBuilder and StringBuffer - first one is more efficient, but we can't use it in multithreading constructions and should use StringBuffer.
Let's finalize: 
If we create single string - use new String() constructor.
If we create string in cycle - StringBuilder
if it's cycle in multiple threads - StringBuffer